tools/protocol-schema-check/src/main.rs

```diff
@@ -0,0 +1,114 @@
+#![allow(unused_imports)]
+/// Needed for `inventory`, helps to register all `ProtocolStruct`-ed structs.
+/// This is needed to be able to iterate over all structs and check their
+/// hashes.
+/// Iteration is done by `ProtocolStructInfo`s generated by `ProtocolStruct`
+/// macro.
+// TODO (#11755): add to CI.
+use near_primitives::*;
+
+use near_structs_checker_core::ProtocolStructInfo;
+use std::any::TypeId;
+use std::collections::hash_map::DefaultHasher;
+use std::collections::{BTreeMap, HashMap, HashSet};
+use std::fs;
+use std::hash::{Hash, Hasher};
+use std::path::Path;
+
+fn compute_hash(
+    info: &ProtocolStructInfo,
+    structs: &HashMap<TypeId, &'static ProtocolStructInfo>,
+) -> u32 {
+    let mut hasher = DefaultHasher::new();
```

Comment:
This is probably why your hashes aren't stable. Hashmap hashing introduces some randomness to prevent DoS.

While at that, for the sake of sanity I would outright replace anything `Hash*` with `BTree*` everywhere in this implementation. It would give e.g. sorted error messages by definition.